autorest/azure/auth/authfile.go

```diff
@@ -0,0 +1,124 @@
+package auth
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"unicode/utf16"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/dimchansky/utfbom"
+)
+
+// Auth includes authentication details for ARM clients
+type Auth struct {
+	Authorizer *autorest.BearerAuthorizer
```

Comment:
I'd still like this to be directly exposed, but other than that, this PR looks a lot better! Thanks for entertaining me, @mcardosos!